contents/product-engineers/llm-product-metrics.md

```diff
@@ -0,0 +1,195 @@
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1715070752/posthog.com/contents/blog/generation-count-light.png"
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1715070931/posthog.com/contents/blog/user-feedback-dark.png"
+  alt="Line chart showing total LLM generation and trace count over time"
+/>
+
+## Debug metrics
+
+### Average number of tokens per request
+
+**What is it?** The sum of input and output tokens consumed in each request. Also known as request size.
+
+**Why it's useful:** Essential for optimizing performance and cost. Larger requests lead to higher costs and longer processing times, while smaller requests may lead to more frequent requests.
+
+**Questions to ask**
+- How has the average changed over time?
+- Are there specific features or types of interactions that typically involve larger or smaller requests?
+- How do variations in request size impact the latency and costs of responses?
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1715072339/posthog.com/contents/blog/total-tokens-light.png"
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1715072338/posthog.com/contents/blog/total-tokens-dark.png"
+  alt="Line chart showing average number of tokens per LLM generation"
+/>
+
+### Generation latency
+
+**What is it?** The time it takes for the LLM to generate a response.
+
+**Why it's useful:** Helps identify performance bottlenecks and ensures your UX meets user expectations for speed.
+
+**Questions to ask**
+
+- Are there specific types of requests that have higher latency, and what can be done to address them?
+- Are there latency spikes, and what causes them?
+- How does latency correlate with user satisfaction and retention rates?
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1715072494/posthog.com/contents/blog/latency-light.png"
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1715072493/posthog.com/contents/blog/latency-dark.png"
+  alt="Line chart showing generation latency per LLM request"
+/>
+
+### Error rate
+
+**What is it?** The percentage of requests that result in errors.
+
+**Why it's useful:** Enables you to pinpoint problematic LLM requests and API calls.
+
+**Questions to ask**
+- What is the overall error rate, and how does it break down into user-facing versus backend errors?
+- What percentage of errors are caused by timeouts?
+- Are there specific features or scenarios that are more prone to errors?
+
+ <ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1715073420/posthog.com/contents/blog/Screenshot_2024-05-07_at_10.16.49_AM.png"
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1715073421/posthog.com/contents/blog/Screenshot_2024-05-07_at_10.16.37_AM.png"
+  alt="Line chart showing total number of LLM errors"
+/> 
+
+## Further reading
+
+- [PostHog for LLMs docs](/docs/llm-analytics) 
```

Comment:
Was planning on merging this into Joe's PR with the new docs page, but looks like that is going to take longer than expected now.
```suggestion
- [PostHog LLM docs](docs/product-analytics/llms) 
```